src/toplevel/learnocaml_toplevel_output.ml

```diff
@@ -180,6 +180,46 @@ let output_html ?phrase output html =
   Js_utils.Manip.appendChild output.container div ;
   insert output ?phrase (Html (html, div)) div
 
+
+(* Module to generate new id *)
+module Id_generator : sig
+  val get_fresh_id : unit -> int
+  val reset_ids : unit -> unit
+end =
+struct
+  let id = ref 0
+
+  let get_fresh_id () =
+    let idx = !id in
+    id := !id + 1;
+    idx
+
+  let reset_ids () =
+    id := 0
+end
+
+let replace_markup idx markup svg =
```

Comment:
Please add some comments here to explain the purpose of these functions: you should explain the problem they solve and how they do solve it.